src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift

```diff
@@ -306,7 +306,12 @@ final class HTTPClient: Client {
 				}
 			case let .failure(error):
 				Log.error("Failed to authorize OTP due to error: \(error).", log: .api)
-				completion(.failure(.invalidResponseError))
+				switch error {
+				case .noNetworkConnection:
+					completion(.failure(.noNetworkConnection))
+				default:
+					completion(.failure(.invalidResponseError))
+				}
 			}
 		})
 	}
```

Comment:
It would be great if you can add a new unit test to the test file of the authorize call in which you test the new error case

Reply:
Nice!